tests/shell/tests.py

```diff
@@ -23,25 +25,84 @@ class ShellCommandTestCase(SimpleTestCase):
 
     def test_command_option(self):
         with self.assertLogs("test", "INFO") as cm:
-            call_command(
-                "shell",
-                command=(
-                    "import django; from logging import getLogger; "
-                    'getLogger("test").info(django.__version__)'
-                ),
-            )
+            with captured_stdout():
+                call_command(
+                    "shell",
+                    command=(
+                        "import django; from logging import getLogger; "
+                        'getLogger("test").info(django.__version__)'
+                    ),
+                )
         self.assertEqual(cm.records[0].getMessage(), __version__)
 
     def test_command_option_globals(self):
         with captured_stdout() as stdout:
-            call_command("shell", command=self.script_globals)
+            call_command("shell", command=self.script_globals, verbosity=0)
         self.assertEqual(stdout.getvalue().strip(), "True")
 
     def test_command_option_inline_function_call(self):
         with captured_stdout() as stdout:
-            call_command("shell", command=self.script_with_inline_function)
+            call_command("shell", command=self.script_with_inline_function, verbosity=0)
         self.assertEqual(stdout.getvalue().strip(), __version__)
 
+    @override_settings(INSTALLED_APPS=["shell"])
+    def test_no_settings(self):
+        test_environ = os.environ.copy()
+        if "DJANGO_SETTINGS_MODULE" in test_environ:
+            del test_environ["DJANGO_SETTINGS_MODULE"]
+        for verbosity, expected_stdout in [
+            ("0", ""),
+            (
+                "1",
+                "Auto-imports are disabled as settings are not configured.\n"
+                "DJANGO_SETTINGS_MODULE value is None.\n"
+                "HINT: Ensure that the settings module is configured and set.\n\n",
+            ),
+        ]:
```

Comment:
I would also test verbosity 2:
```suggestion
        error = (
            "Automatic imports are disabled since settings are not configured.\n"
            "DJANGO_SETTINGS_MODULE value is None.\n"
            "HINT: Ensure that the settings module is configured and set.\n\n",
        )
        for verbosity, expected_stdout in [("0", ""), ("1", error), ("2", error)]:
```

Reply:
Nice! I just realized my verbosity 0 asserts were poor so I have slightly tweaked the way I check for the errors